### PR TITLE
IP address validation

### LINKED ISSUE #371: consider validating ip4 / ip6 addresses on set
currently when creating a jail, we can currently create it with a perfectly invalid IP:

```
# ioc create --basejail --name tpl-web ip4_addr="vtnet0|.2/16"
tpl-web successfully created from 11.1-RELEASE!
```

only on start can we see that something may be off:

```
# ioc start tpl-web
[-] JailLaunch@tpl-web: FAILED [0.135s]: 2
Launching jail tpl-web failed
*** Error code 1
```

but not quite what.

we may want to validate this.

### PR DESCRIPTION
**Pull request by gronke**
fixes #371 

- Uses Python ipaddress module

### CODE REVIEW

File: iocage/lib/Config/Jail/Properties/Addresses.py [hunk 1]
```diff
@@ -43,27 +53,35 @@ def __init__(
...
+
+    def __parse_address(
+        self,
+        address: IPAddressInput
+    ) -> typing.Union[ipaddress.IPv4Interface, ipaddress.IPv6Interface]:
+        if isinstance(address, str) is True:
+            address = address.lower()
+            if address in ["accept_rtadv", "dhcp"]:
+                return address
+        return ipaddress.ip_interface(address)
```
Comment (igalic) [inline iocage/lib/Config/Jail/Properties/Addresses.py]: i'm confused about this function name… but i looked it up and it makes perfect sense

but what we now don't actually handle is the [ValueError, which ip_interface raises](https://docs.python.org/3/library/ipaddress.html#ipaddress.ip_interface) if it encounters an error.

File: iocage/lib/Config/Jail/Properties/Addresses.py [hunk 2]
```diff
@@ -154,6 +169,37 @@ def add(
...
+    ) -> None:
+
+        if isinstance(addresses, list) is False:
+            self.add(nic=nic, addresses=[addresses], notify=notify)
+            return
+
+        _addresses: typing.List[IPAddressInput] = []
+        try:
+            _addresses = [self.ADDRESS_CLASS(address) for address in addresses]
+        except ipaddress.AddressValueError as e:
```
Comment (igalic) [inline iocage/lib/Config/Jail/Properties/Addresses.py]: i think this should simply be ValueError, at least according to the documentation

**Comment by igalic** [comment]
looking at Travis Errors now, i only now realize how very work-in-progress this whole thing is ;_)

**Comment by gronke** [comment]
When I label a PR as "Work in Progress" you're looking at bare drafts. We can discuss the direction, but please do not expect them to be functional before the label disappears.

In this case I had to pause my work because a couple of bugs needed attention first. 🐞

File: iocage/lib/Config/Jail/Properties/Addresses.py (hunk 1 above)
Comment (gronke) [reply iocage/lib/Config/Jail/Properties/Addresses.py]: There are three kinds of address-types supported - Host, Net and Interface.

- Host: may not include a netmask
- Net: may only refer to the network address + netmask
- Interface: stores full client address `<IP>/<mask>`

File: iocage/lib/Config/Jail/Properties/Addresses.py (hunk 2 above)
Comment (gronke) [reply iocage/lib/Config/Jail/Properties/Addresses.py]:
```
>>> import ipaddress
>>> ipaddress.IPv6Interface("192.168.23.1/24")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/ipaddress.py", line 2077, in __init__
    IPv6Address.__init__(self, addr[0])
  File "/usr/local/lib/python3.6/ipaddress.py", line 1917, in __init__
    self._ip = self._ip_int_from_string(addr_str)
  File "/usr/local/lib/python3.6/ipaddress.py", line 1654, in _ip_int_from_string
    raise AddressValueError(msg)
ipaddress.AddressValueError: At least 3 parts expected in '192.168.23.1'
```

**Comment by gronke** [comment]
@igalic the branch is updated and the WIP flag removed. 🤓

**Pull request merged**